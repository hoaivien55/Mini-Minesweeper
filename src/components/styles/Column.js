import styled from 'styled-components';

const Column = styled.div`
    flex-grow: ${(props) => props.flexGrow || "1"};
    display: flex;
    justify-content: ${(props) => props.justifyContent || "center"};
    align-items: ${(props) => props.alignItems || "center"};
`;

export default Column;