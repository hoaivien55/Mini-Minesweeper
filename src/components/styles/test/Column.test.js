import renderer from "react-test-renderer";
import Column from "../Column";

it("Column: Virtual DOM snapshot", () => {
	const tree = renderer.create(<Column />).toJSON();
	expect(tree).toMatchSnapshot();
});

it("Column: change props", () => {
	const tree = renderer
		.create(<Column flexGrow="2" justifyContent="start" alignItems="right" />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
