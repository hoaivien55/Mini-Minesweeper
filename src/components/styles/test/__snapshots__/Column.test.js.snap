// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column: Virtual DOM snapshot 1`] = `
.c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
/>
`;

exports[`Column: change props 1`] = `
.c0 {
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: right;
  -webkit-box-align: right;
  -ms-flex-align: right;
  align-items: right;
}

<div
  className="c0"
/>
`;
